src/playdate/sound.nim

```diff
@@ -132,12 +134,38 @@ proc `offset=`*(this: FilePlayer, offset: float32) =
     privateAccess(PlaydateSoundFileplayer)
     playdate.sound.fileplayer.setOffset(this.resource, offset.cfloat)
 
+proc privateFilePlayerFinishCallback(soundSource: SoundSourcePtr, userdata: pointer) {.cdecl, raises: [].} =
+    try:
+        let filePlayer = cast[FilePlayer](userdata)
+        if filePlayer.finishCallback != nil:
+            filePlayer.finishCallback(filePlayer)
+    except:
+        echo "Error while calling FilePlayer finish callback"
```

Comment:
We might want to add `getCurrentExceptionMsg()` here

Reply:
Removed the echo, not needed